src/main/java/com/ericsson/gerrit/plugins/eiffel/configuration/EiffelPluginConfiguration.java

```diff
@@ -46,7 +62,71 @@ public EiffelPluginConfiguration(final String pluginName, final NameKey project,
             throw new ExceptionInInitializerError(String.format("Can't read %s plugin configuration for project %s: %s",
                     pluginName, project.toString(), e.getMessage()));
         }
+        // Read plugin configuration
+        this.enabled = pluginConfig.getBoolean(ENABLED, false);
+        this.filter = pluginConfig.getString(FILTER);
+        this.remremPublishURL = pluginConfig.getString(REMREM_PUBLISH_URL);
+        this.remremUsername = pluginConfig.getString(REMREM_USERNAME);
+        this.remremPassword = pluginConfig.getString(REMREM_PASSWORD);
+        //flow context is optional
+        this.flowContext = pluginConfig.getString(FLOW_CONTEXT);
 
+        // No point to check other config parameters if plugin is disabled
+        boolean isValidated = validatePluginConfig(pluginName, project);
+        if (!isValidated) {
+            return;
+        }
         LOGGER.info("Loaded plugin configuration: {}", pluginConfig.toString());
     }
+
+    /**
+     * This method validates the Eiffel Configurations.
+     * @param pluginName
+     * @param project
+     */
+    private boolean validatePluginConfig(final String pluginName, final NameKey project) {
+        if (!this.enabled) {
+            return false;
+        }
+        
+        // Make sure that REMReM configuration is defined, otherwise we won't be able to send messages.
+        if (this.remremPublishURL == null) {
+            throw new ExceptionInInitializerError(
```

Comment:
Is this caught somewhere? Maybe it's not needed, i don't know exactly how the plugin works.

Reply:
This is needed when the plugin is enabled and the configurations are not set. The plugin will not load and fail.

Reply:
This has been moved. This will not start the plugin if it is empty.

Reply:
I didnt mean that the check isn't needed. I meant, is it needed to catch the Exception, because it was not cought somewhere. :) I understand the confusion though because i was not clear enough